app/main.py

```diff
@@ -2,0 +1,30 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.__class__.alive.append(self)
```

Comment:
Appending `self` to `self.__class__.alive` means that all instances of `Animal` and its subclasses will share the same `alive` list. If you want each subclass to maintain its own list, consider using an instance variable or a different approach.